platform/viewer/src/lib/DICOMSR/handleSR.js

```diff
@@ -0,0 +1,63 @@
+import * as dcmjs from 'dcmjs';
+import OHIF from '@ohif/core';
+import retrieveDataFromSR from './retrieveDataFromSR';
+import retrieveDataFromMeasurements from './retrieveDataFromMeasurements';
+
+import { api } from 'dicomweb-client';
+
+const retrieveMeasurementFromSR = async series => {
+  // TODO: Find a way to get the server and get the wadoRoot url
+  const url = 'https://server.dcmjs.org/dcm4chee-arc/aets/DCM4CHEE/rs';
+  const config = {
+    url,
+    headers: OHIF.DICOMWeb.getAuthorizationHeader(),
+  };
+
+  const dicomWeb = new api.DICOMwebClient(config);
+
+  const instance = series.getFirstInstance();
+  const options = {
+    studyInstanceUID: instance.getStudyInstanceUID(),
+    seriesInstanceUID: instance.getSeriesInstanceUID(),
+    sopInstanceUID: instance.getSOPInstanceUID(),
+  };
+
+  return dicomWeb.retrieveInstance(options).then(retrieveDataFromSR);
+};
+
+const stowSRFromMeasurements = async measurements => {
+  const dataset = retrieveDataFromMeasurements(measurements);
```

Comment:
In order to prevent repeating the same measurements previous series in subsequent series it might be best to filter out the measurements that have already been saved. I have this simple filter that seems to do the job in my testing. It might be the wrong place for this though and there might be a better way to check the origin/dirty state of the measurement.

```
  // Exclude the measurements that are already stored
  // TODO detect whether those measurements were modified
  measurements.allTools = measurements.allTools.filter( (m) => { if (m.imageId) { return false;} return true; });
```

Reply:
We actually should be picking up only the latest SR series into the study, that should contain the latest state of saved measurements. Somehow, the series are being full filled 1 by 1, so in this case, everytime this code runs, it goes thourgh all SRs and store it again in the viewer. 
I am currently working on this.